cmd/mt-control-server/handlers.go

```diff
@@ -0,0 +1,339 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/Shopify/sarama"
+	"github.com/gocql/gocql"
+	"github.com/grafana/globalconf"
+	"github.com/grafana/metrictank/api/models"
+	"github.com/grafana/metrictank/api/response"
+	"github.com/grafana/metrictank/cmd/mt-control-server/controlmodels"
+	"github.com/grafana/metrictank/expr/tagquery"
+	"github.com/grafana/metrictank/idx"
+	"github.com/grafana/metrictank/schema"
+	"github.com/grafana/metrictank/schema/msg"
+	"github.com/grafana/metrictank/util"
+	log "github.com/sirupsen/logrus"
+	"github.com/tinylib/msgp/msgp"
+	"gopkg.in/macaron.v1"
+)
+
+var metrictankUrl string
+var maxDefsPerMsg int
+var indexConcurrency int
+
+func ConfigHandlers() {
+	FlagSet := flag.NewFlagSet("handlers", flag.ExitOnError)
+
+	FlagSet.StringVar(&metrictankUrl, "metrictank-url", "http://metrictank:6060", "URL for MT cluster")
+	FlagSet.IntVar(&maxDefsPerMsg, "max-defs-per-msg", 1000, "Maximum defs per control message")
+	FlagSet.IntVar(&indexConcurrency, "index-concurrency", 5, "Concurrent queries to run against cassandra (per request)")
+
+	globalconf.Register("handlers", FlagSet, flag.ExitOnError)
+}
+
+func makeLogId() string {
+	logId := "uninit"
+	uuid, err := gocql.RandomUUID()
+	if err == nil {
+		logId = uuid.String()
+	}
+	return logId
+}
+
+func makeMessage(partition int32, op schema.Operation, cm *schema.ControlMsg) (*sarama.ProducerMessage, error) {
+	cm.Op = op
+	var b bytes.Buffer
```

Comment:
since `ReadIndexControlMsg(data []byte) (schema.ControlMsg, error)` lives in the schema package, the reverse code should probably also live there.

the serializing and deserializing of older metric messages also lives in schema/msg/msg.go